source/module_io/berryphase.cpp

```diff
@@ -493,14 +494,18 @@ void berryphase::Macroscopic_polarization(const int npwx,
 	ModuleBase::Vector3<double> rcell_1(GlobalC::ucell.G.e11,GlobalC::ucell.G.e12,GlobalC::ucell.G.e13);
 	ModuleBase::Vector3<double> rcell_2(GlobalC::ucell.G.e21,GlobalC::ucell.G.e22,GlobalC::ucell.G.e23);
 	ModuleBase::Vector3<double> rcell_3(GlobalC::ucell.G.e31,GlobalC::ucell.G.e32,GlobalC::ucell.G.e33);
-	int *mod_ion = new int[GlobalC::ucell.nat];
-	double *pdl_ion_R1 = new double[GlobalC::ucell.nat];
-	double *pdl_ion_R2 = new double[GlobalC::ucell.nat];
-	double *pdl_ion_R3 = new double[GlobalC::ucell.nat];  
-	ModuleBase::GlobalFunc::ZEROS(mod_ion,GlobalC::ucell.nat);
-	ModuleBase::GlobalFunc::ZEROS(pdl_ion_R1,GlobalC::ucell.nat);
-	ModuleBase::GlobalFunc::ZEROS(pdl_ion_R2,GlobalC::ucell.nat);
-	ModuleBase::GlobalFunc::ZEROS(pdl_ion_R3,GlobalC::ucell.nat);
+	//int *mod_ion = new int[GlobalC::ucell.nat];
+	std::vector<int> mod_ion(GlobalC::ucell.nat);
+	//double *pdl_ion_R1 = new double[GlobalC::ucell.nat];
+	std::vector<double> pdl_ion_R1(GlobalC::ucell.nat);
+	//double *pdl_ion_R2 = new double[GlobalC::ucell.nat];
+	std::vector<double> pdl_ion_R2(GlobalC::ucell.nat);
+	//double *pdl_ion_R3 = new double[GlobalC::ucell.nat];  
+	std::vector<double> pdl_ion_R3(GlobalC::ucell.nat);
+	ModuleBase::GlobalFunc::ZEROS(mod_ion.data(),GlobalC::ucell.nat);
+	ModuleBase::GlobalFunc::ZEROS(pdl_ion_R1.data(),GlobalC::ucell.nat);
+	ModuleBase::GlobalFunc::ZEROS(pdl_ion_R2.data(),GlobalC::ucell.nat);
+	ModuleBase::GlobalFunc::ZEROS(pdl_ion_R3.data(),GlobalC::ucell.nat);
```

Comment:
No need to do this, please see [this](https://en.cppreference.com/w/cpp/container/vector/vector#notes:~:text=The%20overload%20(4)%20zeroes%20out%20elements%20of%20non%2Dclass%20types%20such%20as%20int)
```suggestion
```

Reply:
But if I don't change it, the program will report an error